core/src/services/onedrive/pager.rs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::mem;
+
+use crate::{
+    raw::{
+        build_rel_path,
+        oio::{self},
+    },
+    EntryMode, Metadata,
+};
+
+use super::graph_model::{GraphApiOnedriveResponse, ItemType};
+use crate::Result;
+use async_trait::async_trait;
+
+pub(crate) struct OnedrivePager {
+    root: String,
+    path: String,
+    onedrive_response: GraphApiOnedriveResponse,
+}
+
+impl OnedrivePager {
```

Comment:
@xuanwo not sure whether this will be a burden for you:

When you add the e2e test, this code may not work with more modification.

Maybe I should run e2e test locally first 😂

Reply:
Yes, we need to setup the behavior tests frist

Reply:
Do you have a tentative timeline?

If I'd like to contribute the list op or other cloud drive support, what should I do first?

Reply:
Maybe you can try testing our behaviour tests locally first?

Reply:
Good idea! Hopefully I will get these done in one day or two!